packages/flutter/lib/src/gestures/tap_and_drag.dart

```diff
@@ -1287,6 +1310,7 @@ sealed class BaseTapAndDragGestureRecognizer extends OneSequenceGestureRecognize
       invokeCallback<void>('onDragEnd', () => onDragEnd!(endDetails));
     }
 
+    _correctedPosition = null;
```

Comment:
This ensures that the corrected (dragged delta) only gets cleared after the end event has been dispatched.